src/test/java/RouterTest.java

```diff
@@ -0,0 +1,53 @@
+import java.util.HashMap;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertTrue;
+import org.junit.Before;
+import org.junit.Test;
+
+public class RouterTest {
+  private HashMap<String, Handler> routes; 
+  private Router router;
+  
+  @Before 
+  public void setup() {
+    this.routes = new HashMap<String, Handler>();
+    String rootPath = System.getProperty("user.dir");
+    String jarDirectoryPath = rootPath + "/build/libs";
+    this.routes.put("/", new RootHandler(jarDirectoryPath));
+
+    this.router = new Router(routes);
+  }
+  
+  @Test 
+  public void returnsNotFoundHandlerWhenURIIsNotFound() {
+    Request request = new Request.Builder()
+                                 .method("GET")
+                                 .uri("/path/that/does/not/exist")
+                                 .version("1.1")
+                                 .build();                      
+                                 
+    String expectedResponseString = "HTTP/1.1 404 Not Found\r\n" + 
```

Comment:
There is probably a better way to test the response than creating the exact string (seems like it will be annoying to write these strings as you add routes with complex responses)

Reply:
Right now, it feels like my Router tests are redundantly testing the `Response` builder. Is it enough just to test the status code and/or the reason phrase as evidence that the correct handler was returned? If the URI wasn't found, then we should always expect a 404. If the handler was found, but the method was not allowed, we should expect a 405. Successful requests will have a response in the 200's. I'm not sure if some of the upcoming features will require asserting on parts of the response beyond the status code, but--for now--it seems like the status code alone keeps the test at the proper scope.